### DISCUSSION TITLE
[easy] how to bundle an index.js to a single js file

### DISCUSSION THREAD

**Discussion by bozdoz** [swc-project/swc#8641] [Q&A]
I think I have a basic build that I can't figure out: I want to re-export relative files from an index.tsx file, like so:

index.tsx
```tsx
export { default as Provider, type Variant } from "./Provider";
```

If I build that, with `swc ./src/index.tsx -o dist/index.js`, using this config:

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": true
    },
    "target": "es5"
  },
  "module": {
    "type": "commonjs"
  },
  "isModule": true
}
```

I get:

```js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "Provider", {
    enumerable: true,
    get: function() {
        return _Provider.default;
    }
});
var _Provider = /*#__PURE__*/ _interop_require_default(require("./Provider"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
```

The **problem** being that the Provider is no where to be seen in this script! Do I have to build every file, or can swc not bundle into a single script?

I *must* be missing something basic, but I have I read though the documentation on the swc config, and looked through issues without finding anything.

**Comment by kwonoj** [comment]
Since SWC is not a bundler, it is expected. Use your preferred bundler for the work.
